test/AlgebraicGeometry/Schemes/WeilDivisor.jl

```diff
@@ -170,3 +171,35 @@ end
   @test intersect(D1, D2) == intersect(D1, D3) == 21
   @test intersect(D2, D3) == 1
 end
+
+@testset "decomposition" begin 
+  P3 = projective_space(QQ, 3)
+  S = homogeneous_coordinate_ring(P3)
+  (x, y, z, w) = gens(S)
+  I = ideal(S, x^2*y^3)
+  II = ideal_sheaf(P3, I)
+  X = covered_scheme(P3)
+  D = weil_divisor(II)
+  E = oscar.irreducible_decomposition(D)
+  @test length(keys(coefficient_dict(E))) == 2
+  @test 2*one(coefficient_ring(E)) in values(coefficient_dict(E))
+  @test 3*one(coefficient_ring(E)) in values(coefficient_dict(E))
+end
+
+@testset "intersection numbers on surfaces" begin
+  P3 = projective_space(QQ, 3)
+  S = homogeneous_coordinate_ring(P3)
+  (x,y, z, w) = gens(S)
+  I = ideal(S, [x^4+y^4+z^4+w^4])
+  II = ideal_sheaf(P3, I)
+  P = covered_scheme(P3)
+  inc = oscar.CoveredClosedEmbedding(covered_scheme(P3), II)
+  X = domain(inc)
+  C1 = EffectiveCartierDivisor(ideal_sheaf(P3, [x+y+z+w]))
+  C2 = EffectiveCartierDivisor(ideal_sheaf(P3, [x^2*y + y^2*z + z^2*w + w^2*x]))
+  C1 = pullback(inc)(C1)
+  C2 = pullback(inc)(C2)
+  d = intersect(weil_divisor(C1), weil_divisor(C2))
+  pts = oscar.irreducible_decomposition(intersect(C1, C2))
```

Comment:
The first line computes without using the glueings. The second one proceeds via the general method and first writes `C1` as an `AlgebraicCycle` with irreducible `components`. In particular, this uses `maximal_associated_points` and thus needs glueings.